src/app/courses/courses.service.ts

```diff
@@ -1,19 +1,29 @@
 import { Injectable } from '@angular/core';
 import { CouchService } from '../shared/couchdb.service';
 import { Subject } from 'rxjs/Subject';
+import { PlanetMessageService } from '../shared/planet-message.service';
+import { UserService } from '../shared/user.service';
 
 // Service for updating and storing active course for single course views.
 @Injectable()
 export class CoursesService {
 
   course: any = { _id: '' };
+  private shelfUpdate = new Subject<any[]>();
```

Comment:
there is already observable for shelfupdate on user service we can just listen to that service. So it is not required to create new listener